# HG changeset patch
# User Marcel J.E. Mol <marcel@mesa.nl>
# Date 1372331222 -3600
# Node ID 554358946d979a097692777dc43a5cdce49f7e00
# Parent 7291f552074a1cfac946e1cb42d5405239e813fb
pygrub/GrubConf: fix boot problem for fedora 19 grub.cfg (2nd attempt)

Booting a fedora 19 domU failed because a it could not properly
parse the grub.cfg file. This was cased by

	set default="${next_entry}"

This statement actually is within an 'if' statement, so maybe it would
be better to skip code within if/fi blocks...
But this patch seems to work fine.

Signed-off-by: Marcel Mol <marcel@mesa.nl>
Acked-by: Ian Campbell <ian.campbell@citix.com>
Tested-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>

diff -r 7291f552074a tools/pygrub/src/GrubConf.py
--- a/tools/pygrub/src/GrubConf.py
+++ b/tools/pygrub/src/GrubConf.py
@@ -429,6 +429,8 @@ class Grub2ConfigFile(_GrubConfigFile):
                         arg = "0"
                     elif arg.strip() == "${next_entry}":
                         arg = "0"
+                    elif arg.strip() == "${next_entry}":
+                        arg = "0"
                     setattr(self, self.commands[com], arg.strip())
                 else:
                     logging.info("Ignored directive %s" %(com,))
